### PR TITLE
Make clone more resilient to a CouchDB service connection timeouts

### PR DESCRIPTION
**Pull request by svujic**
More info about this patch can be found at https://stackoverflow.com/questions/47055971/how-to-copy-a-couchdb-in-dokku/47193504
However sometimes a available CouchDB service  responds with {"error":"unauthorized","reason":"Name or password is incorrect."}.
This patch doesn't  solve this problem and I'm not aware what causes this error.

### CODE REVIEW

File: subcommands/clone [hunk 1]
```diff
@@ -22,8 +22,28 @@ service-clone-cmd() {
...
+  attemptcount=0
+  R=2
+  succ_str=' Successfully.'
+  until [[ $R = 0 || $R = 1 ]]; do
+    attemptcount=$((attemptcount+1))
+    STDOUT1=$(service_export "$SERVICE" | service_import "$NEW_SERVICE" 2>&1) || true
+    if [[ ! "$STDOUT1" = *"${succ_str}" ]]; then
+      if [ $attemptcount = $attempts ]; then
+        R=1
+        echo -e "\nERROR: CouchDB Import failed - Stopping\n"
```
Comment (josegonzalez) [inline subcommands/clone]: Can you use the log helpers for this instead?

**Comment by josegonzalez** [comment]
@svujic can you see if we can just fix the underlying script here: https://github.com/danielebailo/couchdb-dump

I think that would be a better fix than trying to paper over it above, especially as we might end up partially importing the dataset multiple times.

**Comment by svujic** [comment]
We can fix underlying script but we will ended with STDOUT1 string that we must validate for success. Because we pipe export to import directly.
It's all just a timeout problem in majority of cases (service isn't ready at that time).
Maybe we can try to export some small data set (dummy data) until service start to respond and then make real export?

**Comment by josegonzalez** [comment]
The script should just retry if it detects an error, and if we're unable to export the data, then thats fine, the clone should fail/we should delete the old service. I think thats the appropriate thing to do here. What this PR does is somewhat paper over the issue and can potentially leave the new clone in a bad state.

**Comment by svujic** [comment]
In accordance with your advice I fixed underlying couchdb-dump script. Then I simplified last commit. Please take a look.

File: subcommands/clone [hunk 2]
```diff
@@ -22,8 +22,13 @@ service-clone-cmd() {
 
   service_create "$NEW_SERVICE" "${@:3}"
   dokku_log_info1 "Copying data from $SERVICE to $NEW_SERVICE"
-  service_export "$SERVICE" | service_import "$NEW_SERVICE" > /dev/null 2>&1 || true
+
+  succ_str=' Successfully.'
+  STDOUT1=$(service_export "$SERVICE" | service_import "$NEW_SERVICE" 2>&1) || true
+  if [[ ! "$STDOUT1" = *"${succ_str}" ]]; then
```
Comment (josegonzalez) [inline subcommands/clone]: Can we instead look at the exit code?

**Pull request closed without merging**